nandtool/nandtool.c

```diff
@@ -193,32 +296,74 @@ static void nandtool_help(const char *prog)
 }
 
 
+static int nandtool_parseRange(const char *str, int *start, int *size)
+{
+	char *endptr;
+
+	*start = strtoul(str, &endptr, 0);
+	if (*endptr == '\0') {
+		/* Default case, one block */
+		*size = 1;
+	}
+	else if (*endptr == ':') {
+		str = endptr + 1;
+		*size = strtoul(str, &endptr, 0);
+		if ((str == endptr) || (*endptr != '\0')) {
+			return -1;
+		}
+	}
+	else {
+		return -1;
+	}
+
+	return 0;
+}
+
+
 int main(int argc, char **argv)
 {
 	int check = 0, raw = 0, flash_start = -1, erase_start = -1, erase_size = -1, write_cleanmarkers = 0;
-	char *dev, *tok, *path = NULL;
+	int dump_start = -1, dump_size = -1, oob = 0;
+	const char *ipath = NULL, *opath = NULL;
+	char *dev;
 	int c, err;
 
 	if (isatty(STDOUT_FILENO))
 		nandtool_common.interactive = 1;
 
-	while ((c = getopt(argc, argv, "e:i:r:s:chjq")) != -1) {
+	while ((c = getopt(argc, argv, "e:d:i:o:r:s:bchjq")) != -1) {
```

Comment:
Please fix the `'-r'` option by removing the `':'` - this option doesn't take any arguments.